Calculator.c

```diff
@@ -0,0 +1,219 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MaxLen 100
+int Error = 0;
+
+int IsSpace(char Character)
+{
+    return Character == ' ' || Character == '\t' || Character == '\n';
+}
+
+void RemoveWhiteSpaces(char *Expression)
+{
+    char *Left = Expression;
+    char *Right = Expression;
+    while (*Right != '\0')
+    {
+        if (!IsSpace(*Right))
+        {
+            *Left++ = *Right;
+        }
+        Right++;
+    }
+    *Left = '\0';
+}
+
+int IsDigit(char Character)
+{
+    return Character >= '0' && Character <= '9';
+}
+
+int StringCompare(const char *str1, const char *str2)
+{
+    while (*str1 != '\0' && *str2 != '\0')
+    {
+        if (*str1 != *str2)
+        {
+            return (unsigned char)(*str1) - (unsigned char)(*str2);
+        }
+        str1++;
+        str2++;
+    }
+    return (unsigned char)(*str1) - (unsigned char)(*str2);
+}
+
+int StringLength(const char *String)
+{
+    int Length = 0;
+    while (String[Length] != '\0')
+    {
+        Length++;
+    }
+    return Length;
+}
+
+int Precedence(char Operator)
+{   
+    int Value = 0;
+    if (Operator == '*' || Operator == '/')
+        Value = 2;
+    if (Operator == '+' || Operator == '-')
+        Value = 1;
+    return Value;
+}
+
+int PerformOperation(int OperandA, int OperandB, char Operator)
+{   
+    int Result = 0;
+    switch (Operator)
+    {
+    case '/':
+        if (OperandB == 0)
+        {
+            printf("Error: Division by zero.\n");
+            Error = 1;
+            break;
+        }   
+        Result = OperandA / OperandB;
+        break;  
+
+    case '*':
+        Result = OperandA * OperandB;
+        break;
+
+    case '+':
+        Result = OperandA + OperandB;
+        break;
+
+    case '-':
+        Result = OperandA - OperandB;
+        break;
+
+    default:
+        printf("Error: Invalid operator.\n");
+        Error = 1;
+    }
+    return Result;
+}
+
+int Calculate(char *Expression)
+{
+    char Operators[MaxLen];
+    int ValuesStack[MaxLen];
+    int ValuesTop = -1, OperatorsTop = -1;
+
+    for (int Index = 0; Expression[Index] != '\0'; Index++)
+    {   
+        if (Error) break;
+        if (IsDigit(Expression[Index]))
```

Comment:
IsDigit is a built in function, try using different logic

Reply:
I have Custom Written IsDigit() Function in the Program which follows Pascal Case.
Wheres inbuilt function isdigit().